src/main/java/com/amazon/dlic/auth/ldap2/LDAPConnectionFactoryFactory.java

```diff
@@ -127,17 +130,29 @@ public ConnectionPool createConnectionPool() {
         }
 
         result.setValidator(getConnectionValidator());
+
+        LDAPConnectionFactoryFactory.checkForDeprecatedSetting(settings, ConfigConstants.LDAP_LEGACY_POOL_PRUNING_PERIOD, ConfigConstants.LDAP_POOL_PRUNING_PERIOD);
+        LDAPConnectionFactoryFactory.checkForDeprecatedSetting(settings, ConfigConstants.LDAP_LEGACY_POOL_IDLE_TIME, ConfigConstants.LDAP_POOL_IDLE_TIME);
           
         result.setPruneStrategy(new IdlePruneStrategy(
-            Duration.ofMinutes(this.settings.getAsLong(ConfigConstants.LDAP_POOL_PRUNING_PERIOD, this.settings.getAsLong("pruning.period", 5l))),
-            Duration.ofMinutes(this.settings.getAsLong(ConfigConstants.LDAP_POOL_IDLE_TIME, this.settings.getAsLong("pruning.idleTime", 10l))))
+            Duration.ofMinutes(this.settings.getAsLong(ConfigConstants.LDAP_POOL_PRUNING_PERIOD, this.settings.getAsLong(ConfigConstants.LDAP_LEGACY_POOL_PRUNING_PERIOD, 5l))),
+            Duration.ofMinutes(this.settings.getAsLong(ConfigConstants.LDAP_POOL_IDLE_TIME, this.settings.getAsLong(ConfigConstants.LDAP_LEGACY_POOL_IDLE_TIME, 10l))))
         );
 
         result.initialize();
 
         return result;
     }
 
+    /**
+     * Checks for an deprecated key found in a setting, logs that it should be replaced with the another key
+     */
+    private static void checkForDeprecatedSetting(final Settings settings, final String legacySettingKey, final String validSettingKey) {
```

Comment:
Would it make sense to move this function into Settings and pass the logger in so it can be re-used in other places?

Reply:
Settings definitions themselves have a way to do this, see [Setting.java](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/Setting.java#L130).

As these legacy settings not registered into the schema when the plugin is registered we need a mechanism like I made here to support it.

Reply:
There are unregistered settings outside of LDAP as well. See https://github.com/opensearch-project/security/blob/main/src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java#L60 as an example. Does it make sense to move this in a utility class in `org.opensearch.security.support` in case we need to support this for other use cases in the future?

Reply:
I would like to merge the code as is, in the future we could organize if we reuse it.  What do you think?

Reply:
I see another ticket in the backlog where this function could be applicable (it just so happens to also be related to LDAP): https://github.com/opensearch-project/security/issues/2059

Its functionally fine to have the method here, but I see an opportunity for re-use that we may miss in the future. @DarshitChanpura @cliu123 thoughts?

Reply:
I agree with the idea of making it more generic so it can be re-used as needed